include/mppic/motion_models.hpp

```diff
@@ -38,62 +65,111 @@ class MotionModel
     }
   }
 
+  /**
+   * @brief Whether the motion model is holonomic, using Y axis
+   * @return Bool If holonomic
+   */
   virtual bool isHolonomic() = 0;
-  virtual void applyConstraints(models::ControlSequence & /*state*/) {}
+
+  /**
+   * @brief Apply hard vehicle constraints to a control sequence
+   * @param control_sequence Control sequence to apply constraints to
+   */
+  virtual void applyConstraints(models::ControlSequence & /*control_sequence*/) {}
 };
 
+/**
+ * @class mppi::AckermannMotionModel
+ * @brief Ackermann motion model
+ */
 class AckermannMotionModel : public MotionModel
 {
 public:
+  /**
+    * @brief Constructor for mppi::AckermannMotionModel
+    */
   explicit AckermannMotionModel(ParametersHandler * param_handler)
   {
     auto getParam = param_handler->getParamGetter("AckermannConstraints");
     getParam(min_turning_r_, "min_turning_r", 0.2);
   }
 
+  /**
+   * @brief Whether the motion model is holonomic, using Y axis
+   * @return Bool If holonomic
+   */
   bool isHolonomic() override
   {
     return false;
   }
 
+  /**
+   * @brief Apply hard vehicle constraints to a control sequence
+   * @param control_sequence Control sequence to apply constraints to
+   */
   void applyConstraints(models::ControlSequence & control_sequence) override
   {
     auto & vx = control_sequence.vx;
     auto & wz = control_sequence.wz;
 
-    auto view = xt::masked_view(wz, vx / wz > min_turning_r_);
-    view = xt::sign(vx) / min_turning_r_;
+    auto view = xt::masked_view(wz, xt::fabs(vx) / xt::fabs(wz) > min_turning_r_);
```

Comment:
Note bug fix here

Reply:
Prevented any actual penalties for ackermann steering due to not considering the vx itself (just the sign, and the sign of the wrong thing I believe)